client/extensions/woocommerce/state/wc-api/product-categories/actions.js

```diff
@@ -40,3 +49,22 @@ export function fetchProductCategoriesSuccess( siteId, data ) {
 	};
 }
 
+function isValidCategoriesArray( categories ) {
+	for ( let i = 0; i < categories.length; i++ ) {
+		if ( ! isValidProductCategory( categories[ i ] ) ) {
+			// Short-circuit the loop and return now.
+			return false;
+		}
+	}
+	return true;
+}
+
+function isValidProductCategory( category ) {
+	return (
+		category &&
+		category.id && ( typeof category.id ) === 'number' &&
```

Comment:
This syntax seems odd.  Did you mean to do this instead?

`category.id && ( 'number' === typeof category.id ) &&`

Same comment for the next two lines